packages/amplify-cli/src/extensions/amplify-helpers/invoke-plugin-method.ts

```diff
@@ -0,0 +1,35 @@
+import { $TSContext } from 'amplify-cli-core';
+
+export const invokePluginMethod = async <T>(
+  context: $TSContext,
+  category: string,
+  service: string | undefined,
```

Comment:
I was thinking about it, and had to look for places when we have it anywhere and the order is like this because of what the function does. the first 3 parameters are grouped for `getCategoryPluginInfo` and then the params are for the method invocation, so the order seems to be more logical like this.

Reply:
I think I'd still vote for putting it at the end of the args list, but I do understand the logic for the ordering. I'll let it be your call.

Reply:
I wondering if its better to just get the method and leave the invocation part to the caller

Reply:
@edwardfoyle @yuth I was thinking about both solutions to pack the argument and just return the method, at this point both seems to be irrelevant to the problem the PR solves Today. Longer term having an interface that we can rely on that a method is present and some info in the manifest or else that guarantees that a given plugin does not violate that contract seems to be better. Returning the method is another possibility, but that's a different indirection. I leave it now as it is, we can refactor it later with platform changes, like a "provider" model as we did for functions. Plugin provides `IAMPolicyProvider` and such.